kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/LogReplay.java

```diff
@@ -182,7 +187,7 @@ public long getVersion() {
 
   /** Returns the crc info for the current snapshot if the checksum file is read */
   public Optional<CRCInfo> getCurrentCrcInfo() {
-    return currentCrcInfo;
+    return cachedCrcInfo.filter(crc -> crc.getVersion() == logSegment.getVersion());
```

Comment:
Also -- to clarify: the consumers of this API only ever want the CRCInfo if it was read during log replay? They never just want the CRCINfo for the latest version, even if it wasn't read during log replay? That semantic seems a bit odd to me -- can you please clarify / explain?

Reply:
the consumer just try to get crcInfo of the current version. We return if it is cached. Updating the cache is not necessary during log replay's construction.

Reply:
> the consumer just try to get crcInfo of the current version.

But this API will _only_ return it if it was read / cached?

My confusion is this:
- it seems like consumers want the CRC info for the current version. full stop. if it exists, they want it
- this API returns it if and only if it exists **and we had to read it to update the snapshot hint**

So, there seems to be a gap?

Reply:
So the concern here is --- when version hint is used, then crcInfo will always empty. This was unfortunately the behavior before, although problematic.

I updated with a fix. (Previously it will cause CRC missing when snapshot hint is used, now it is getting worse as we load domain metadata, so agree to fix it along the way.)